platforms/software/dependency-management/src/main/java/org/gradle/internal/resolve/resolver/DefaultVariantArtifactResolver.java

```diff
@@ -100,8 +108,7 @@ private ResolvedVariant createResolvedVariant(
         ImmutableArtifactTypeRegistry artifactTypeRegistry,
         ImmutableList<? extends ComponentArtifactMetadata> artifacts
     ) {
-        ImmutableAttributes attributes = artifactTypeRegistry.mapAttributesFor(artifactVariant.getAttributes(), artifacts);
-
+        ImmutableAttributes attributes = attributeDesugaring.desugar(artifactTypeRegistry.mapAttributesFor(artifactVariant.getAttributes(), artifacts));
```

Comment:
I think we should do the desugaring in `ResolvedArtifactCollectingVisitor`, after any internal processing, and just before we present the artifact to the user.

Reply:
Made this change.